Geometry_Engine/Modify/SplitAtPoints.cs

```diff
@@ -50,41 +50,146 @@ public static List<Arc> SplitAtPoints(this Arc arc, List<Point> points, double t
 
             List<Arc> result = new List<Arc>();
             List<Point> cPts = new List<Point>();
-            Vector normal = arc.Normal();
 
+            double sqTol = tolerance * tolerance;
+            Plane arcPlane = arc.FitPlane();
+            //Collect all points on the circle of the arc (points not on arc filtered out lower down)
+            //Doing this half manually (Not calling IsOnCurve) due to wanting the projected points anyway, and this is more efficient.
+            //Angles checked to be in domain while looping through the points further down the algorithm
             foreach (Point point in points)
             {
-                if (point.IsOnCurve(arc, tolerance))
-                    cPts.Add(point);
-            }
+                Point projPt = point.Project(arcPlane);
+                double projSqDist = projPt.SquareDistance(point);
+                if (projSqDist > sqTol)
+                    continue;   //Not in plane
 
-            cPts.Add(arc.StartPoint());
-            cPts.Add(arc.EndPoint());
-            cPts = cPts.CullDuplicates(tolerance);
-            cPts = cPts.SortAlongCurve(arc, tolerance);
+                double inPlaneDist = Math.Abs(projPt.Distance(arcPlane.Origin) - arc.Radius);
```

Comment:
Because the radius need to be subtracted.

`Math.Abs(projPt.SquareDistance(arcPlane.Origin) - arc.Radius * arc.Radius);`
Is not the same thing as 

`Math.Pow(Math.Abs(projPt.Distance(arcPlane.Origin) - arc.Radius), 2);`

unfortunately. 

Had a thought about it (and agree on the line 70 comment) but think this is the simplest and quickest it can be done.